From 395f8aabfecb28820006d37ec37e9ffe1d2eb1e3 Mon Sep 17 00:00:00 2001
From: Phillip Susi <psusi@ubuntu.com>
Date: Fri, 11 May 2018 11:01:01 -0400
Subject: [PATCH 90/92] Fix atari label false positives

The atari label gets false positives easily, so probe it after
all other labels have said no.
---
 libparted/libparted.c | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/libparted/libparted.c b/libparted/libparted.c
index d855d0e..3bd071d 100644
--- a/libparted/libparted.c
+++ b/libparted/libparted.c
@@ -80,12 +80,13 @@ extern void ped_disk_atari_init ();
 static void
 init_disk_types ()
 {
+	/* Note that probing is done in the reverse order of init */
 	ped_disk_loop_init ();	/* must be last in the probe list */
 
 #if defined __s390__ || defined __s390x__
 	ped_disk_dasd_init();
 #endif
-
+	ped_disk_atari_init (); /* easy false positives, so probe others first */
 	ped_disk_sun_init ();
 #ifdef ENABLE_PC98
 	ped_disk_pc98_init ();
@@ -97,7 +98,6 @@ init_disk_types ()
 	ped_disk_bsd_init ();
 	ped_disk_amiga_init ();
 	ped_disk_aix_init ();
-	ped_disk_atari_init ();
 }
 
 extern void ped_file_system_amiga_init (void);
-- 
2.17.1

